core/src/main/java/smile/classification/DataFrameClassifier.java

```diff
@@ -27,7 +27,7 @@
  *
  * @author Haifeng Li
  */
-public interface DataFrameClassifier {
+public interface DataFrameClassifier extends SoftClassifier<Tuple> {
```

Comment:
This is wrong. They are orthogonal.

Reply:
ok I will remove it but then I am stuck I am afraid because of the problem above